civis/resources/_resources.py

```diff
@@ -476,3 +507,22 @@ def generate_classes(api_key, user_agent, api_version="1.0", resources="base"):
     raw_swagger = get_swagger_spec(api_key, user_agent, api_version)
     swagger = JsonRef.replace_refs(raw_swagger)
     return parse_swagger(swagger, api_version, resources)
+
+
+def generate_classes_maybe_cached(cache, api_key, user_agent, api_version,
+                                  resources):
+    """Generate class objects either from /endpoints or a local cache."""
+    if cache is None:
+        classes = generate_classes(api_key, user_agent, api_version, resources)
+    else:
+        if isinstance(cache, OrderedDict):
+            raw_spec = cache
+        elif isinstance(cache, str):
+            with open(cache, "r") as f:
+                raw_spec = json.load(f, object_pairs_hook=OrderedDict)
+        else:
+            msg = "cache must be an OrderedDict or str, given {}"
+            raise ValueError(msg.format(type(cache)))
+        spec = JsonRef.replace_refs(raw_spec)
+        classes = parse_swagger(spec, api_version, resources)
```

Comment:
`lru_cache` won't be able to memoize this since `collections.OrderedDict` isn't hashable.  Unless you feel strongly, I think it's not worth rolling our own memoization function.